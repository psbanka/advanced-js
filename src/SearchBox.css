.SearchBox {
  color: red;
  border-width: 1px;
  border-style: solid;
  border-color: green;
}
