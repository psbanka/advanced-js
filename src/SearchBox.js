/**
 * This is a module that provides the top-level search-box
 */

import React, { Component } from 'react'
// import './SearchBox.css'

export default class SearchBox extends Component {

  render () {
    const textChangeCallback = this.props.onFilterTextInput.bind(this)
    const checkboxChangeCallback = this.props.onFilterCheckBoxInput.bind(this)
    return (
      <div className='SearchBox'>
        <button style={{color: 'red'}}>Confirm</button>
        <input
          id='in-stock-textbox'
          type='text'
          onChange={textChangeCallback}
          placeholder='Search...'
          value={this.props.searchTerm} />
        <p>
          <input
            id='in-stock-checkbox'
            onClick={checkboxChangeCallback}
            checked={this.props.inStock}
            type='checkbox' />
          Only show products in stock
        </p>
      </div>
    )
  }
}
