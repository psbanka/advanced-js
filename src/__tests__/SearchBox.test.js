import React from 'react'
import ReactDOM from 'react-dom'
import SearchBox from '../SearchBox'
import { shallow, mount, render } from 'enzyme'
import sinon from 'sinon';
import renderer from 'react-test-renderer';

/* global it describe */

describe('SearchBox', () => {
  let onFilterTextInput
  let onFilterCheckBoxInput

  beforeEach(() => {
    onFilterTextInput = sinon.spy();
    onFilterCheckBoxInput = sinon.spy();
  })

  it('SearchBox snapshot check', () => {
    const component = renderer.create(
      <SearchBox
        onFilterTextInput={onFilterTextInput}
        onFilterCheckBoxInput={onFilterCheckBoxInput}
      />
    )
    const json = component.toJSON()
    console.log(json)
    expect(json).toMatchSnapshot()
  })

  it('will make proper callback when adding search text', () => {
    const searchBox = shallow(
      <SearchBox
        onFilterTextInput={onFilterTextInput}
        onFilterCheckBoxInput={onFilterCheckBoxInput}
      />
    )
    const textBox = searchBox.find('#in-stock-textbox')
    textBox.simulate('change', {target: {value: 'ball'}})
    expect(onFilterTextInput.calledOnce).toBe(true);
    expect(onFilterTextInput.callCount).toBe(1);
  })
})
