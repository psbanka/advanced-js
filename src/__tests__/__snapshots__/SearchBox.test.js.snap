exports[`SearchBox does a snapshot check 1`] = `
<div
  className="SearchBox">
  <button
    style={
      Object {
        "color": "red",
      }
    }>
    Confirm
  </button>
  <input
    id="in-stock-textbox"
    onChange={[Function]}
    placeholder="Search..."
    type="text"
    value={undefined} />
  <p>
    <input
      checked={undefined}
      id="in-stock-checkbox"
      onClick={[Function]}
      type="checkbox" />
    Only show products in stock
  </p>
</div>
`;
